src/sentry/static/sentry/app/components/modals/sentryAppPublishRequestModal.tsx

```diff
@@ -0,0 +1,155 @@
+    const scopeQuestionPlainText = `${scopeQuestionBaseText}${scopes.join(', ')}.`;
+
+    const scopeLabel = (
+      <span>
+        {scopeQuestionBaseText}
+        {scopes.map((scope, i) => (
+          <React.Fragment key={i}>
+            {i > 0 && ', '} <code>{scope}</code>
+          </React.Fragment>
+        ))}
+        .
+      </span>
+    );
+
+    //No translations since we need to be able to read this email :)
+    const baseFields = [
+      {
+        type: 'textarea',
+        required: true,
+        label: 'What does your integration do? Please be as detailed as possible.',
+        autosize: true,
+        rows: 1,
+        inline: false,
+      },
+      {
+        type: 'textarea',
+        required: true,
+        label: 'What value does it offer customers?',
+        autosize: true,
+        rows: 1,
+        inline: false,
+      },
+      {
+        type: 'textarea',
+        required: true,
+        label: scopeLabel,
+        autosize: true,
+        rows: 1,
+        inline: false,
+        meta: scopeQuestionPlainText,
+      },
+      {
+        type: 'textarea',
+        required: true,
+        label: 'Do you operate the web service your integration communicates with?',
+        autosize: true,
+        rows: 1,
+        inline: false,
+      },
+    ];
+    //dynamically generate the name based off the index
+    return baseFields.map((field, index) =>
+      Object.assign({name: `question${index}`}, field)
+    );
+  }
+
+  handleSubmitSuccess = () => {
+    addSuccessMessage(t('Request to publish %s successful.', this.props.app.slug));
+    this.props.closeModal();
+  };
+
+  handleSubmitError = () => {
+    addErrorMessage(t('Request to publish %s fails.', this.props.app.slug));
+  };
+
+  render() {
+    const {app} = this.props;
+    const endpoint = `/sentry-apps/${app.slug}/publish-request/`;
+    const forms = [
+      {
+        title: t('Questions to answer'),
+        fields: this.formFields,
+      },
+    ];
+    return (
+      <React.Fragment>
+        <Header>{t('Publish Request Questionnaire')}</Header>
+        <Body>
+          <Explanation>
+            {t(
+              `Please fill out this questionnaire in order to get your integration evaluated for publication.
+              Once your integration has been approved, users outside of your organization will be able to install it.`
+            )}
+          </Explanation>
+          <Form
+            allowUndo
+            apiMethod="POST"
+            apiEndpoint={endpoint}
+            onSubmitSuccess={this.handleSubmitSuccess}
+            onSubmitError={this.handleSubmitError}
+            model={this.form}
+            submitLabel={t('Request Publication')}
+            onCancel={() => this.props.closeModal()}
+          >
+            <JsonForm location={this.props.location} forms={forms} />
+          </Form>
+        </Body>
+      </React.Fragment>
+    );
+  }
+}
+
+const Explanation = styled.div`
```

Comment:
We should use the functional format so that it's consistent between native DOM els and custom components

```suggestion
const Explanation = styled('div')`
```